torchtune/datasets/_chat.py

```diff
@@ -141,7 +142,7 @@ def chat_dataset(
         tokenizer=tokenizer,
         source=source,
         convert_to_messages=convert_to_messages,
-        chat_format=_get_chat_format(chat_format),
+        chat_format=_get_chat_format(chat_format) if chat_format else None,
```

Comment:
This check and the check in the class are technically incorrect.

`if arg` checks for a boolean or a co-erced empty list. 

What you want is to check `if arg is None`